include/crud-terms.php

```diff
@@ -293,54 +293,17 @@ public function set_pre_term_name( $name ) {
 	 * @return string Slug with a language suffix if found.
 	 */
 	public function set_pre_term_slug( $slug, $taxonomy ) {
-		if ( ! $this->model->is_translated_taxonomy( $taxonomy ) ) {
-			return $slug;
-		}
-
-		if ( ! $slug ) {
-			$slug = sanitize_title( $this->pre_term_name );
-		}
+		$term_slug = new PLL_Term_Slug( $this->model, $slug, $taxonomy, $this->pre_term_name );
 
-		if ( ! term_exists( $slug, $taxonomy ) ) {
-			return $slug;
+		if ( ! $term_slug->can_add_suffix() ) {
+			return $term_slug->get_slug();
 		}
 
-		/**
-		 * Filters the subsequently inserted term language.
-		 *
-		 * @since 3.3
-		 *
-		 * @param PLL_Language|null $lang     Found language object, null otherwise.
-		 * @param string            $taxonomy Term taonomy.
-		 * @param string            $slug     Term slug
-		 */
-		$lang = apply_filters( 'pll_inserted_term_language', null, $taxonomy, $slug );
-
-		if ( ! $lang instanceof PLL_Language ) {
-			return $slug;
-		}
-
-		$parent = 0;
-		if ( is_taxonomy_hierarchical( $taxonomy ) ) {
-			/**
-			 * Filters the subsequently inserted term parent.
-			 *
-			 * @since 3.3
-			 *
-			 * @param int          $parent   Parent term ID, 0 if none.
-			 * @param string       $taxonomy Term taxonomy.
-			 * @param string       $slug     Term slug
-			 */
-			$parent = apply_filters( 'pll_inserted_term_parent', 0, $taxonomy, $slug );
-		}
-
-		$term_id = (int) $this->model->term_exists_by_slug( $slug, $lang, $taxonomy, $parent );
-
 		// If no term exist in the given language with that slug, it can be created.
-		if ( ! $term_id ) {
-			$slug .= '-' . $lang->slug;
+		if ( ! $term_slug->get_term_id() ) {
+			return $term_slug->get_suffixed_slug( '-' );
 		}
 
-		return $slug;
+		return $term_slug->get_slug();
```

Comment:
Reading the code, `get_term_id()` and `get_suffixed_slug()` seem useless to me now.
We could internalize the logic even more by removing these method and keeping only `get_slug()` (with an extra parameter for the suffix).
And on Polylang Pro we could use `get_slug()` without any parameter depending on its specific condition (with all the globals).
To go a bit further, i'd rename `get_slug()` into  get()` since "slug" is already in the class name. I think this is something we tend to do nowadays.

Reply:
Something like this https://github.com/polylang/polylang/pull/1490/commits/109de5cefd885a2b12ec37dbc120792f8f36d307 ?

Reply:
I think I still need the `$term_id` on the PLL Pro side to be able to compare it with the global `ID`.
See https://github.com/polylang/polylang-pro/blob/3.6.2/modules/share-slug/share-term-slug.php#L224.

Reply:
Indeed, let's keep the ID getter and use it only in Polylang Pro then. Is that ok for you?